src/DependencyInjection/DoctrineMigrationsExtension.php

```diff
@@ -135,6 +135,13 @@ public function load(array $configs, ContainerBuilder $container): void
         $container->setParameter('doctrine.migrations.preferred_em', $config['em']);
         $container->setParameter('doctrine.migrations.preferred_connection', $config['connection']);
 
+        if ($config['connections'] !== []) {
+            $filterDefinition = $container->getDefinition('doctrine_migrations.schema_filter_listener');
+            foreach ($config['connections'] as $connection) {
```

Comment:
Rather than looping on this, how about creating a compiler pass that uses this parameter: https://github.com/doctrine/DoctrineMigrationsBundle/blob/e858ce0f5c12b266dce7dce24834448355155da7/src/DependencyInjection/CompilerPass/ConfigureDependencyFactoryPass.php#L82

Reply:
You're right, it makes much more sense. I have simplified my fix.